src/openrct2-ui/windows/StaffList.cpp

```diff
@@ -531,6 +532,30 @@ class StaffListWindow final : public Window
             auto intent = Intent(WindowClass::Peep);
             intent.PutExtra(INTENT_EXTRA_PEEP, staff);
             ContextOpenIntent(&intent);
+
+            // If autoposition of staff is disabled, set the pickup tool on the newly created staff window
```

Comment:
This looks fine. I need to have a little think if the fact the window is being opened by an intent is an issue with this function checking for a window being open

Reply:
Had a think with @ZehMatt and it would be best to do this differently. If we move the intent to open into a branch that is just for auto position we can have another intent to open purely from the pickup game action. Would look something like:
```
            auto* staff = GetEntity<Staff>(actionResult.StaffEntityId);

            // If autoposition of staff is disabled, pickup peep then open the staff window
            if (staff->State == PeepState::Picked)
            {
                picked_peep_old_x = staff->x;
                CoordsXYZ nullLoc{};
                nullLoc.SetNull();

                PeepPickupAction pickupAction{ PeepPickupType::Pickup,
                                               EntityId::FromUnderlying(staff->sprite_index.ToUnderlying()), nullLoc,
                                               NetworkGetCurrentPlayerId() };
                pickupAction.SetCallback(
                    [&staff](const GameAction* ga, const GameActions::Result* result) {
                        if (result->Error != GameActions::Status::Ok)
                            return;

                        auto intent = Intent(WindowClass::Peep);
                        intent.PutExtra(INTENT_EXTRA_PEEP, staff);
                        auto* wind = ContextOpenIntent(&intent);
                        if (wind != nullptr)
                        {
                            ToolSet(*wind, WC_STAFF__WIDX_PICKUP, Tool::Picker);
                        }
                    });
                GameActions::Execute(&pickupAction);
            }
            else
            {
                // Open window for new staff.
                auto intent = Intent(WindowClass::Peep);
                intent.PutExtra(INTENT_EXTRA_PEEP, staff);
                ContextOpenIntent(&intent);
            }
```

Reply:
Done!